test/adapter/json_api/key_case_test.rb

```diff
@@ -312,187 +294,156 @@ def test_success_document_key_transform_camel_lower
-          result = adapter.serializable_hash(@options)
-
-          expected_errors_object =
-            { :Errors =>
+          result = serialize_resource(@error_resource, options(:camel, serializer: ActiveModel::Serializer::ErrorSerializer))
+          expected_errors_object = {
+            Errors:
               [
-                { :Source => { :Pointer => '/data/attributes/published_at' }, :Detail => 'must be in the future' },
-                { :Source => { :Pointer => '/data/attributes/title' }, :Detail => 'must be longer' }
+                {
+                  Source: { Pointer: '/data/attributes/published_at' },
+                  Detail: 'must be in the future'
+                },
+                {
+                  Source: { Pointer: '/data/attributes/title' },
+                  Detail: 'must be longer'
+                }
               ]
           }
           assert_equal expected_errors_object, result
         end
 
         def test_error_document_key_transform_camel_lower
-          mock_request(:camel_lower)
+          result = serialize_resource(@error_resource, options(:camel_lower, serializer: ActiveModel::Serializer::ErrorSerializer))
+          expected_errors_object = {
+            errors: [
+              {
+                source: { pointer: '/data/attributes/published_at' },
+                detail: 'must be in the future'
+              },
+              {
+                source: { pointer: '/data/attributes/title' },
+                detail: 'must be longer'
+              }
+            ]
+          }
+          assert_equal expected_errors_object, result
+        end
 
-          resource = ModelWithErrors.new
-          resource.errors.add(:published_at, 'must be in the future')
-          resource.errors.add(:title, 'must be longer')
+        private
 
-          serializer = ActiveModel::Serializer::ErrorSerializer.new(resource)
-          adapter = ActiveModelSerializers::Adapter::JsonApi.new(serializer)
-          result = adapter.serializable_hash(@options)
+        def options(key_transform = nil, extra_options = {})
+          context = Minitest::Mock.new
+          context.expect(:request_url, URI)
+          context.expect(:query_parameters, {})
+          context.expect(:key_transform, key_transform)
+          context.expect(:url_helpers, Rails.application.routes.url_helpers)
+          {
+            serialization_context: context,
+            adapter: :json_api
+          }.merge(extra_options)
+        end
 
-          expected_errors_object =
-            { :errors =>
-              [
-                { :source => { :pointer => '/data/attributes/published_at' }, :detail => 'must be in the future' },
-                { :source => { :pointer => '/data/attributes/title' }, :detail => 'must be longer' }
-              ]
-          }
-          assert_equal expected_errors_object, result
+        def serialize_resource(resource, options)
```

Comment:
since the method calls `as_json`, should really be `resource_serialization`. Also, good to be consistent and use the test helper: `serializable(resource, options).as_json`